crates/via-router/src/path.rs

```diff
@@ -13,7 +12,7 @@ pub(crate) enum Pattern {
 /// path separated by `/`.
 pub(crate) struct PathSegments<'a> {
     pub(crate) value: &'a str,
-    segments: SmallVec<[(usize, usize); 6]>,
+    segments: Vec<(usize, usize)>,
```

Comment:
Using `SmallVec` only makes the router between 10 and 20 nanoseconds faster. Considering that the URL path is unbounded user input, I think using the Vec from the standard library provides safety guarantees that make this a reasonable tradeoff.